presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.file;
+
+import at.favre.lib.crypto.bcrypt.BCrypt;
+import at.favre.lib.crypto.bcrypt.IllegalBCryptFormatException;
+import com.google.common.base.Splitter;
+import com.google.common.io.BaseEncoding;
+import io.airlift.log.Logger;
+import io.prestosql.spi.PrestoException;
+
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.List;
+
+import static io.prestosql.spi.StandardErrorCode.CONFIGURATION_INVALID;
+
+public class EncryptionUtil
+{
+    private static final Logger log = Logger.get(EncryptionUtil.class);
+
+    private EncryptionUtil() {}
+
+    // enforce minumum cost in for BCrypt Passwords : Min value default: 8
+    public static boolean checkFormatBCryptPassword(String password, int limit)
+    {
+        try {
+            BCrypt.HashData hashData = BCrypt.Version.VERSION_2A.parser.parse(password.getBytes());
+            return hashData.cost >= limit;
+        }
+        catch (IllegalBCryptFormatException e) {
+            log.error("Error while checking for bcrypt password format " + e);
```

Comment:
Make this throw an error directly. In Presto, we avoid the log-and-throw pattern, since that leads to the same error being logged in multiple places in different ways, which is confusing. So either log or throw, but not both.

Reply:
Done:
` try {
            BCrypt.HashData hashData = BCrypt.Version.VERSION_2A.parser.parse(password.getBytes());
            return hashData.cost >= minCost;
        }
        catch (IllegalBCryptFormatException e) {
            throw new PrestoException(CONFIGURATION_INVALID, "Invalid bcrypt Cost: %s (The minimum allowed BCrypt Cost should be 8)");
        }`